snyk_feed.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+# Copyright © 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Author: Yusuf Zainee <yzainee@redhat.com>
+#
+
+"""Script which fetches the feed from snyk and stores it in S3 bucket."""
+
+import os
+import jwt
+import time
+import requests
+import logging
+from datetime import datetime
+from graph_snyk_cve_sync import Helper
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class SnykDataFetcher:
+    """Snyk data class to fetch the information from Snyk."""
+
+    def __init__(self):
+        """Init function to add default values."""
+        self.helper = Helper()
+        self.URL = os.environ.get('SNYK_URL', '')
+        self.ISS = os.environ.get('SNYK_ISS', '')
+        self.TOKEN = os.environ.get('SNYK_TOKEN', '')
+
+    def validate(self):
+        """Validate method to see all the params are set."""
+        try:
+            assert self.URL
+            assert self.ISS
+            assert self.TOKEN
+        except AssertionError:
+            raise ValueError('Snyk Feed fetch aborted. Either URL, ISS or TOKEN missing')
+        logger.info("Snyk credentials validated.")
+        return True
+
+    def fetch(self):
+        """Fetch function to fetch the feed from Snyk."""
+        payload = {
+            'iss': self.ISS,
+            'iat': int(time.time())
+        }
+        jwt_token = jwt.encode(
+            payload,
+            self.TOKEN,
+            algorithm='HS256'
+        )
+        headers = {
+            'Authorization': jwt_token
+        }
+        try:
+            res = requests.get(self.URL, headers=headers)
+        except Exception as e:
+            logger.error('Unable to get snyk feed. Reason: %r' % e)
+        logger.info("Snyk feed successfully fetched. Storing in S3.")
+        self.store_data(res.text)
+        return True
+
+    def store_data(self, data):
+        """Store the snyk data into the respective S3 bucket."""
+        utc_now = datetime.utcnow()
+        self.helper.store_json_content(data, "snyk-feed/" + utc_now.strftime('%d-%m-%y') + ".json")
+
+    def disable_snyk_run(self, date):
+        """Enable or disable snyk run."""
```

Comment:
New function Docs should be: 
"""
Enable or Disable Synk Feed Sync run. 
:return: 
    - True, if its Saturday OR Current Time is not on 12AM utc.
    - False, SNYK_INGESTION_FORCE is set to False 
"""